### DISCUSSION TITLE
Prioritise panda-css styles over chakra-ui styles to progressively migrate a component library

### DISCUSSION THREAD

**Discussion by mayteio** [chakra-ui/panda#2018] [Q&A] [answered]
# Context

We've developed a second generation component library using `panda-css`. Our first-gen uses `@chakra-ui/react`. We've got a monster app to migrate over, so we're trying to do this progressively by changing the components used on a feature-by-feature basis (replacing around 700 `import { xyz } from "our-old-chakra-lib"`) progressively. 

When I try to import components that use `panda-css`, the chakra-ui styles for selectors like `button`, I have to `!important` many styled-props like `color`.

# Questions
- Is there a way to make the `import "./global.css"` styles take priority over the `chakra-ui` styles that are injected in the `<head />` tag of the page? Would ordering the imports after `<head><link ... /></head>`s generated by `chakra-ui`?
- Has anyone done this before?
- Can you offer any advice on a migration path here that won't leave us with 1000s of `!` declarations?

**Comment by astahmer** [comment]
- you can try using this postcss plugin to remove the `@layer` (and still emulate their specificity)
https://www.npmjs.com/package/@csstools/postcss-cascade-layers
- on the `ChakraProvider` you can set `resetCSS={false}` to remove the global styles from Chakra that affects your whole app (or use the `resetScope` to specify a selector)
- you could generate that CSS reset on your side and include it in the same `@layer` (reset) as Panda
https://gist.github.com/astahmer/1154453909b1d4b6b6ce06aad4eace89

---

btw, since you're coming from Chakra, here's a Panda preset that mirrors the base chakra theme, maybe that can help you migrate some parts
https://github.com/astahmer/crepe-ui/tree/main/packages/preset-chakra

**Reply by mayteio** [reply]
Thanks @astahmer! Great info. I will go through each of these and report back on what was successful.

1. emulate - sounds complicated and prone to error
2. `resetScope` - sounds most promising. Add this, and add the class to every component (rather than reset at the page level). Does the chakra reset differ greatly from the panda reset?
3. I guess I can generate and diff the chakra reset against the panda reset

---

Thanks for the preset! We're already well down the rabbit hole, we probably should have started with that 🫠

**Reply by mayteio** [reply, marked as answer]
Hey thanks for the suggestions! `@csstools/postcss-cascade-layers` just worked automagically by adding it to the `postcss.config.cjs` alongside the panda plugin:
```
const path = require("path");

module.exports = {
  plugins: {
    "@pandacss/dev/postcss": {
      configPath: path.resolve(__dirname, "../../libs/styled-system/panda.config.ts"),
    },
    "@csstools/postcss-cascade-layers": {}
  },
};
```

Thanks so much!
